src/olympia/amo/management/commands/compress_assets.py

```diff
@@ -0,0 +1,302 @@
+import hashlib
+import os
+import re
+import shutil
+import time
+import urllib2
+import uuid
+
+from subprocess import PIPE, call, check_output
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from olympia.lib.jingo_minify_helpers import get_media_root, get_path
+
+
+def path(*args):
+    return os.path.join(get_media_root(), *args)
+
+
+class Command(BaseCommand):
+    help = ('Compresses css and js assets defined in settings.MINIFY_BUNDLES')
+    requires_model_validation = False
+    do_update_only = False
+
+    checked_hash = {}
+    bundle_hashes = {}
+
+    missing_files = 0
+    minify_skipped = 0
+    cmd_errors = False
+    ext_media_path = os.path.join(get_media_root(), 'external')
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--use-uuid', action='store_true', dest='use_uuid',
+            help='Use a uuid as the build id instead of git.')
+        parser.add_argument(
+            '-u', '--update-only', action='store_true', dest='do_update_only',
+            help='Updates the hash only')
+        parser.add_argument(
+            '-t', '--add-timestamp', action='store_true', dest='add_timestamp',
+            help='Add timestamp to hash')
+
+    def generate_build_id(self, use_uuid):
+        if use_uuid:
+            return uuid.uuid4().hex[:8]
+        else:
+            root = getattr(settings, 'JINGO_MINIFY_ASSETS_GIT_ROOT', '.')
+            git_bin = getattr(settings, 'GIT_BIN', 'git')
+            return check_output(
+                [git_bin, '-C', root, 'rev-parse', '--short', 'HEAD']).strip()
+
+    def update_hashes(self, update=False):
+        if update:
+            # Adds a time based hash on to the build id.
+            self.build_id = '%s-%s' % (
+                self.build_id, hex(int(time.time()))[2:])
+
+        build_id_file = os.path.realpath(os.path.join(settings.ROOT,
+                                                      'build.py'))
+        with open(build_id_file, 'w') as f:
+            f.write('BUILD_ID_CSS = "%s"\n' % self.build_id)
+            f.write('BUILD_ID_JS = "%s"\n' % self.build_id)
+            f.write('BUILD_ID_IMG = "%s"\n' % self.build_id)
+            f.write('BUNDLE_HASHES = %s\n' % self.bundle_hashes)
+
+    def handle(self, **options):
+        self.build_id = self.generate_build_id(options.get('use_uuid', False))
+
+        if options.get('do_update_only', False):
+            self.update_hashes(update=True)
+            return
+
+        jar_path = (os.path.dirname(__file__), '..', '..', 'bin',
+                    'yuicompressor-2.4.7.jar')
+        self.path_to_jar = os.path.realpath(os.path.join(*jar_path))
+
+        self.v = '-v' if options.get('verbosity', False) == '2' else ''
+
+        cachebust_imgs = getattr(settings, 'CACHEBUST_IMGS', False)
```

Comment:
Shall we just drop this?  The setting is `True` everywhere.

Reply:
Also done in the new PR